### PR TITLE
tag os3 to rhel for openshift jenkins image 3.6 branch rhel image tes…

### PR DESCRIPTION
**Pull request by gabemontero**
…ting (need rhel with subscriptions)

See discussion between @jupierce , @bparees , and myself in https://github.com/openshift/jenkins/pull/479 including the resolution plan in https://github.com/openshift/jenkins/pull/479#issuecomment-357998316

@openshift/sig-continuous-delivery fyi / ptal

### CODE REVIEW

File: sjb/config/test_cases/test_branch_jenkins_images.yml [hunk 1]
```diff
@@ -16,5 +16,8 @@ extensions:
       repository: "jenkins"
       timeout: 7200
       script: |-
+        # this docker tag is for our 3.6 branch; we did not introduce
+        # the use of the ose image until 3.7
+        docker tag registry.access.redhat.com/openshift3/ose:v3.6 registry.access.redhat.com/rhel7
```
Comment (bparees) [inline sjb/config/test_cases/test_branch_jenkins_images.yml]: i'd be more inclined to do this in the same place where the fake ose:v3.6 image is being created, so someone doesn't go change/remove that logic and break this.

**Comment by gabemontero** [comment]
On Tue, Jan 16, 2018 at 11:47 AM, Ben Parees <notifications@github.com>
wrote:

> *@bparees* commented on this pull request.
> ------------------------------
>
> In sjb/config/test_cases/test_branch_jenkins_images.yml
> <https://github.com/openshift/aos-cd-jobs/pull/998#discussion_r161817447>:
>
> > @@ -16,5 +16,8 @@ extensions:
>        repository: "jenkins"
>        timeout: 7200
>        script: |-
> +        # this docker tag is for our 3.6 branch; we did not introduce
> +        # the use of the ose image until 3.7
> +        docker tag registry.access.redhat.com/openshift3/ose:v3.6 registry.access.redhat.com/rhel7
>
> i'd be more inclined to do this in the same place where the fake ose:v3.6
> image is being created, so someone doesn't go change/remove that logic and
> break this.
>
how about a comment in that file, but the actual tagging in this file, so
as the minimize where the tagging occurs ?

**Comment by bparees** [comment]
> how about a comment in that file, but the actual tagging in this file, so as the minimize where the tagging occurs ?

i'm not sure why minimizing where tagging occurs is a goal.  I would set the goal as minimizing where "constructing the subscribed base images/tags" occurs.  However you're the one most likely to have to maintain this, so it's up to you.

File: sjb/config/test_cases/test_branch_jenkins_images.yml (hunk 1 above)
Comment (gabemontero) [reply sjb/config/test_cases/test_branch_jenkins_images.yml]: ||    how about a comment in that file, but the actual tagging in this file, so as the minimize where ||the tagging occurs ?

| i'm not sure why minimizing where tagging occurs is a goal. I would set the goal as minimizing |where "constructing the subscribed base images/tags" occurs. However you're the one most likely |to have to maintain this, so it's up to you.


I'm worried about a new job getting defined that wants to use `registry.access.redhat.com/rhel7` as its from image and the rhel base job as its parent, and getting tripped up by our tagging.  Is that not a valid concern?

@stevekuznetsov - any opinions ?

Comment (bparees) [reply sjb/config/test_cases/test_branch_jenkins_images.yml]: that new job shouldn't extend the "rhel_base_images" job then. The only purpose of that job is to build images that have subscriptions+necessary repos built in.

(admittedly the job name/description could be better)

**Comment by gabemontero** [comment]
switched location of tag, added clarifying comment

**Comment by gabemontero** [comment]
bump @bparees @openshift/sig-continuous-delivery

**Comment by bparees** [comment]
lgtm

**Comment by bparees** [comment]
@gabemontero are you going to push the job updates to jenkins?

**Comment by gabemontero** [comment]
On Fri, Jan 19, 2018 at 9:14 AM Ben Parees <notifications@github.com> wrote:

> @gabemontero <https://github.com/gabemontero> are you going to push the
> job updates to jenkins?
>
yep i was going to give that a go

I’ll report back with the results

**Comment by gabemontero** [comment]
Hmmm .... I got this:

```
gmontero ~/go/src/github.com/openshift/aos-cd-jobs  (master)$ sjb/push-update-automatic.sh 
[INFO] Checking for existence of push_jenkins_images...
[INFO] Checking for existence of merge_pull_request_jenkins_images...
sjb/push-update.sh: line 20: PASSWORD: unbound variable
sjb/push-update.sh: line 20: PASSWORD: unbound variable
[INFO] Checking for existence of test_branch_jenkins_images...
sjb/push-update.sh: line 20: PASSWORD: unbound variable
[INFO] Checking for existence of test_pull_request_jenkins_images...
sjb/push-update.sh: line 20: PASSWORD: unbound variable
gmontero ~/go/src/github.com/openshift/aos-cd-jobs  (master)$ 
```

Doesn't look right ... any ideas?

**Comment by bparees** [comment]
you need to pass USERNAME and PASSWORD as env vars when you invoke the script.

**Comment by gabemontero** [comment]
duh ;-/

that progressed things, but still no luck:

```
gmontero ~/go/src/github.com/openshift/aos-cd-jobs  (master)$ sjb/push-update-automatic.sh 
[INFO] Checking for existence of merge_pull_request_jenkins_images...
[INFO] Checking for existence of test_branch_jenkins_images...
[INFO] Checking for existence of push_jenkins_images...
[INFO] Checking for existence of test_pull_request_jenkins_images...
[INFO] Creating test_pull_request_jenkins_images...
[INFO] Creating push_jenkins_images...
[INFO] Creating merge_pull_request_jenkins_images...
[INFO] Creating test_branch_jenkins_images...
[ERROR] Failed to create merge_pull_request_jenkins_images: 401
[ERROR] Failed to create push_jenkins_images: 401
[ERROR] Failed to create test_pull_request_jenkins_images: 401
[ERROR] Failed to create test_branch_jenkins_images: 401
gmontero ~/go/src/github.com/openshift/aos-cd-jobs  (master)$ 

```

401 - appears my github ID  is not authorized

**Comment by bparees** [comment]
it's not your github id it wants, it's your jenkins login credentials.

**Comment by gabemontero** [comment]
thanks again ... it worked now

**Pull request merged**